### ISSUE TITLE
How the original RA acted after infiltration

### ISSUE THREAD

**Issue by Snurre86** [OpenRA/OpenRA#12791]
Here is a few images of infiltrated buildings in the original Red Alert. This is how we should have it in missions and as a spectator/in replays. We could also talk about having it this way when playing after double infiltration etc.

![before selecting a infiltrated building](https://cloud.githubusercontent.com/assets/7798635/23072856/9a5c8734-f533-11e6-9576-8cfb5c291ef3.png)
![select it and you see what it is building1](https://cloud.githubusercontent.com/assets/7798635/23072865/a38f967a-f533-11e6-81a6-f29d346ba8a0.png)
![select it and you see what it is building2](https://cloud.githubusercontent.com/assets/7798635/23072870/a79d5b80-f533-11e6-81b4-74058726f0e5.png)
![select it and you see what it is building3](https://cloud.githubusercontent.com/assets/7798635/23073017/4e43575a-f534-11e6-9531-452e81f8ba27.png)

**Comment by pchote** [comment]
Thanks for the info.  This behaviour was deliberately changed as part of making spies more useful in multiplayer, so I don't expect this to change.

**Comment by Snurre86** [comment]
Trying to make them more useful? I can't remember OpenRA being this way at any point. Still useful for replays and specs. And if you could get more spies in the building instead of it being useless it could give you more and more information. like 1 spy in the mcv would give you the building and turret being build (the 2nd image) then 2 spies it would also give you the progress bar etc. I can see many options here.

**Comment by VolkovChitzkoi** [comment]
Just my two cents: I opened some months ago a thread about introducing an idea of a more useful spying, with the ability (in part taken by the RA2 mixing secret units concept) to create particular hidden units in OpenRA (the idea was that, if you used a spy to infiltrate the enemy Radar and enemy Airstrip, you would have access to construct the Nuclear MiG for example, non-playable unit present in a Counterstrike mission). Unfortunatelly this thing got rejected, for there was too much code conditions to add for the "Spy" actors. Unfortunatelly...

**Comment by Snurre86** [comment]
I understand. This feels more like magic for the high tier but it is the low tier that should be prioritized as here is where the player mass is and we have a massive low tier bleed out. Where new players come and play and gets completely squashed and never play again.

**Comment by Smittytron** [comment]
I don’t see how this has anything to do with high tier vs. low tier. When a dev says that spies have been changed to be more useful in multiplayer, it means they are more useful for all players in multiplayer, not just high tier.

**Comment by Snurre86** [comment]
My words have been badly chosen then as only players that have a little more skills use or at least successfully use spies.

**Comment by GraionDilach** [comment]
>Trying to make them more useful? I can't remember OpenRA being this way at any point.

95% of the gameplay differences between WW Red Alert and OpenRA's Red Alert can be summed up as "XY unit became more useful" (great examples are the Thief->Hijacker and the ECM Radar Jammer changes). TBH I also don't see an issue with the current setup. I can imagine that the options should be added to the game (for map/modders) BUT I don't see the need of them having enabled in the RA mod. Also, keep in mind that the production-reveal logic conflicts with OpenRA RA gameplay (AFAIK for the reveal you need to select the infiltrated building, which you can't really do under fog, besides, some of them would be quite redundant anyway, say what's the point of seeing the resource amount when you're just stealing it anyway.)

**Comment by Snurre86** [comment]
All good points, but no chat about spec/replay.

**Comment by pchote** [comment]
Spectators / replays already have a dedicated UI panel for viewing player production.

**Comment by Snurre86** [comment]
The infiltrated building would reveal fog just like it where your own building so you could see the production icon.

**Comment by Snurre86** [comment]
I don't know if there is another issue about it but the UI panel should show the latest tab you used (production, economy, combat, earnings) when reopening it.

**Comment by VolkovChitzkoi** [comment]
Yes, the growing shroud would make the Old Spy "build reveal" logic pretty useless.

**Comment by abcdefg30** [comment]
> The power plant would show 500/800 power or something.

Doesn't it already do that? (#10112)

**Comment by GraionDilach** [comment]
Not to the infiltrator, no. ;)

**Comment by Snurre86** [comment]
Also the spy could steal a little money more money but over a longer time period?

**Comment by Snurre86** [comment]
![openra game_2017_02_21_18_10_37_984](https://cloud.githubusercontent.com/assets/7798635/23225104/00b7ebe4-f931-11e6-8764-7c9214e47b1e.png)
![openra game_2017_02_21_18_10_37_984](https://cloud.githubusercontent.com/assets/7798635/23225035/cb785360-f930-11e6-8b9b-ea84bfe04e51.png)

**Comment by Snurre86** [comment]
This will also help allies see what you are doing by hovering over or selection your buildings.

**Comment by KOYK** [comment]
i like this idea. but instead of regular spy,it will be nicer if the British spy could do that. it adds to the strategic depth of the game,makes the British a more useful nation pick,

And it is used in ares.dll for red alert 2 and it will make an easier choice for modders to add ther red alert 2 mods to the openra engine (which reminds me i have to make a ticket about ares.dll,no one else seams to care about migrating red alert2 mods to openra?)

after all westwood did it for some reason,i think you openra devs must trust them a bit...if the above statement din't do for you.

**Comment by GraionDilach** [comment]
>i have to make a ticket about ares.dll,no one else seams to care about migrating red alert2 mods to openra?

@KOYK You don't have to. I participated in Ares development and I have already migrated my Ares-RA2 mod into OpenRA more-or-less. I can even tell you that while the RA1 infiltration logic was restored in Ares indeed, relatively early (IIRC 0.1/early-0.2 days) NOONE used it, because it didn't worked with PCX/custompaletted cameos until 0.9-0.A or around (And custompaletted cameos were a lot more useful, so much that infact my firest OpenRA PR was to implement that feature).

**Comment by KOYK** [comment]
@GraionDilach well i use it in my mod. btw dint know you ware one of the ares devs,so thank you for your gift to as moders!
